appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleTracker.kt

```diff
@@ -83,20 +88,20 @@ internal class ExperienceLifecycleTracker(
 
     private fun State.track() {
         when (this) {
-            is RenderingStep -> {
-                if (isFirst) {
```

Comment:
BeginningExperience and EndExperience are now working properly as "ExperienceStarted" and "ExperienceEnded"

Reply:
so this was inconsistent with experienceStarted from config.experienceListener ?, I thought for some reason this was old code because both experience_started analytics and the experienceListener should be reporting at the same time. or thats at least what I thought

Reply:
I drew up the this PR state machine design [here](https://whimsical.com/mobile-experience-states-Cb9WqPWhfUjb7tin4t5LYf) and you are right, this is actually changing behavior

It would be nice if we could use BeginningExperience for something though, or else it just seems to be an extra step that does nothing.

Reply:
upon investigation I discovered that the only time we actually report a fatal error is when the PresentationEffect fails to create metadata setting trait, or present(), lets say we are in the first step, this happens BEFORE we get to RenderingStep which is Good. 

but if customers are listening to config.experienceListener, they will receive a experienceStarted, when in fact we never did. 

probably the flag isFirst is needed here, wouldnt that be the case for both the analytics and the listener?